File: app/src/main/java/com/juniperphoton/flipperviewproj/MainActivity.java
package com.juniperphoton.flipperviewproj;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.juniperphoton.flipperviewlib.FlipperView;

public class MainActivity extends AppCompatActivity {
    private FlipperView mFlipperView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mFlipperView = (FlipperView) findViewById(R.id.flipper_view);
        View prevView = findViewById(R.id.prev_btn);
        View nextView = findViewById(R.id.next_btn);

        prevView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mFlipperView.previous();
            }
        });
        nextView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mFlipperView.next();
            }
        });

        for (int i = 0; i < mFlipperView.getChildCount(); i++) {
            final View view = mFlipperView.getChildAt(i);
            view.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mFlipperView.next();
                    Log.d("main", ((Button) v).getText().toString());
                }
            });
        }
    }
}

File: app/src/main/java/com/juniperphoton/flipperviewproj/MainActivity.kt
package com.juniperphoton.flipperviewproj

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.AdapterView
import android.widget.Button
import android.widget.Spinner
import com.juniperphoton.flipperviewlib.FlipperView

class MainActivity : AppCompatActivity() {
    private var flipperView: FlipperView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        flipperView = findViewById(R.id.flipper_view) as FlipperView
        val prevView = findViewById(R.id.prev_btn)
        val nextView = findViewById(R.id.next_btn)

        prevView.setOnClickListener { flipperView!!.previous() }
        nextView.setOnClickListener { flipperView!!.next() }

        (findViewById(R.id.spinner) as Spinner).onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onNothingSelected(parent: AdapterView<*>?) {
            }

            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
                flipperView!!.displayIndex = position
            }
        }

        (0..flipperView!!.childCount - 1)
                .map { flipperView!!.getChildAt(it) }
                .forEach {
                    it.setOnClickListener { v ->
                        flipperView!!.next()
                        Log.d("main", (v as Button).text.toString())
                    }
                }
    }
}
